### PR TITLE
feat(x): Support headers in the fetch tool

### PR DESCRIPTION
**Pull request by fortuna**
I also disable redirects, since that's misleading. We are usually interested in a single fetch.

### CODE REVIEW

File: x/examples/fetch/main.go [hunk 1]
```diff
@@ -89,12 +102,27 @@ func main() {
...
 
 	req, err := http.NewRequest(*methodFlag, url, nil)
 	if err != nil {
 		log.Fatalln("Failed to create request:", err)
 	}
+	for _, headerLine := range headersFlag {
+		name, value, _ := strings.Cut(headerLine, ":")
+		name = strings.TrimSpace(name)
+		value = strings.TrimSpace(value)
+		if name == "" || value == "" {
```
Comment (jyyi1) [inline x/examples/fetch/main.go:121]: [IETF allows header value to contain only spaces (LWS, linear white space)](https://datatracker.ietf.org/doc/html/rfc2616#section-4.2), maybe we just need to check `name`?

**Review by jyyi1 — APPROVED** [review]

File: x/examples/fetch/main.go (hunk 1 above)
Comment (fortuna) [reply x/examples/fetch/main.go:121]: It doesn't, actually
image

Comment (fortuna) [reply x/examples/fetch/main.go:121]: Oh, actually it can have space if it's quoted!

**Review by fortuna — COMMENTED** [review]
Updated the header parsing. I now assume it's a raw header line.

**Review by jyyi1 — APPROVED** [review]

**Pull request merged**